newton/tests/test_equality_constraints.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 The Newton Developers
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import unittest
+
+import warp as wp
+
+import newton
+
+
+class TestEqualityConstraints(unittest.TestCase):
+    def test_multiple_constraints(self):
+        self.sim_time = 0.0
+        self.frame_dt = 1 / 60
+        self.sim_dt = self.frame_dt / 10
+
+        builder = newton.ModelBuilder()
+
+        newton.utils.parse_mjcf(
+            os.path.join(os.path.dirname(__file__), "assets", "constraints.xml"),
+            builder,
+            ignore_names=["floor", "ground"],
+            up_axis="Z",
+            parse_equality=True,
+        )
+
+        self.model = builder.finalize()
+
+        self.solver = newton.solvers.MuJoCoSolver(
+            self.model,
+            use_mujoco=True,
+            solver="newton",
+            integrator="euler",
+            iterations=100,
+            ls_iterations=50,
+            nefc_per_env=100,
+            ncon_per_env=50,
+        )
+
+        self.control = self.model.control()
+        self.state_0, self.state_1 = self.model.state(), self.model.state()
+        newton.sim.eval_fk(self.model, self.model.joint_q, self.model.joint_qd, self.state_0)
+
+        for _ in range(5):
+            for _ in range(5):
+                self.state_0.clear_forces()
+                self.solver.step(self.state_0, self.state_1, self.control, None, self.sim_dt)
+                self.state_0, self.state_1 = self.state_1, self.state_0
+
+            self.sim_time += self.frame_dt
+
+        self.assertGreater(
+            self.solver.mj_model.eq_type.shape[0], 0
+        )  # check if number of equality constraints in mjModel > 0
```

Comment:
I there a way to do any runtime checks on whether the constraints are getting satisfied, e.g. when bodies start out disconnected and are moved in the right configuration based on the equality constraints? Currently the solver stepping doesn't really affect the unit test if you just check the number of constraints.